### DISCUSSION TITLE
Split coloring

### DISCUSSION THREAD

**Discussion by JohnF42** [patrykandpatrick/vico#483] [Ideas]
Hi,
is it possible to do something like this with the new version?

![image](https://github.com/patrykandpatrick/vico/assets/35494414/26e21df7-0a2c-4daf-84f6-3773271c58ec)

P.S. Sorry for my crude drawing ;)

**Comment by patrickmichalik** [comment]
Hello! I’m afraid this isn’t possible yet, but we’ll be looking into adding such functionality.

**Comment by peshkira** [comment]
Do you guys have a roadmap in general for feature requests and features you are working on? 

Additionally, do you have tips, if this can be achieved with a workaround at the moment or any other related hints. Thank you! 🙏

**Reply by Gowsky** [reply]
We don't have any real roadmap, as this is a FOSS project. Our plans change quite often as new issues or requests arise.
Regarding some tips, I'd recommend checking [`TopBottomShader`](https://github.com/patrykandpatrick/vico/blob/770f69ea539cdda1aa32a58e6a6c4ff9006081cf/vico/core/src/main/java/com/patrykandpatrick/vico/core/component/shape/shader/TopBottomShader.kt) which splits shaders across a defined y-axis coordinate. I don't know how you need to determine the y-axis coordinates at which the layers are split, but you'll most likely also need to override `LineCartesianLayer` and `LineSpec`, so the exact y-axis coordinates can be passed to a custom `DynamicShader`. You can see a sample usage of `TopBottomShader` [in Chart 9 in the sample app](https://github.com/patrykandpatrick/vico/blob/dcabdcb50b3600a1e1af7c39994b3b9a877ef803/sample/src/main/java/com/patrykandpatrick/vico/sample/showcase/charts/Chart9.kt#L221). [This is where `LineSpec` receives zero line coordinate](https://github.com/patrykandpatrick/vico/blob/dcabdcb50b3600a1e1af7c39994b3b9a877ef803/vico/core/src/main/java/com/patrykandpatrick/vico/core/chart/layer/LineCartesianLayer.kt#L340). [And here the coordinate is set to `TopBottomShader`](https://github.com/patrykandpatrick/vico/blob/dcabdcb50b3600a1e1af7c39994b3b9a877ef803/vico/core/src/main/java/com/patrykandpatrick/vico/core/chart/layer/LineCartesianLayer.kt#L228).

**Reply by peshkira** [reply]
Thanks for the tips. Will check it out!

> We don't have any real roadmap, as this is a FOSS project. Our plans change quite often as new issues or requests arise.

Yeah, completely understandable. I know it is not easy maintaining a FOSS project and I appreciate the effort you have put in. The library is awesome.

**Reply by patrickmichalik** [reply]
Thanks for the kind words, @peshkira!

**Comment by Gowsky** [comment]
Vico `2.0.0-alpha.26` makes this possible with `AreaFill`. For now, you'll need a custom implementation, but a dedicated factory function for such use cases is planned

**Reply by peshkira** [reply]
Thanks guys. This is soon in scope for my project, so it fits perfectly.

cheers

**Reply by patrickmichalik** [reply]
You’re welcome, @peshkira! We’re glad to hear that.

**Reply by git-vcs** [reply]
Did get this to work @peshkira? I am trying to do something similar and I am struggling a bit with understanding how to make custom implementations.

**Reply by peshkira** [reply]
This got deprioritised a bit, so I haven't yet tried it. But from the api changes I saw, it should be possible to do.
Will ping the thread when I get to it, unless it is resolved before.

**Reply by patrickmichalik** [reply]
@git-vcs, I’d suggest taking a look at how `DoubleAreaFill`—which is instantiated by `AreaFill.double`—is implemented. The definition is found [here](https://github.com/patrykandpatrick/vico/blob/v2.1.0/vico/core/src/main/java/com/patrykandpatrick/vico/core/cartesian/layer/AreaFills.kt#L122-L148). Please note that `DoubleAreaFill` is a subclass of `BaseAreaFill`, where the key logic resides. An `AreaFill` implementation that produces the result from the image above would differ only slightly from `DoubleAreaFill`: instead of a single _y_ value at which the two areas meet, there’d be two separate _y_ values for (1) the bottom edge of the top area and (2) the top edge of the bottom area. Let me know if you have any questions.
